pkg/test/client.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2018 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package test
+
+import (
+	goctx "context"
+	"fmt"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	dynclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type frameworkClient struct {
+	dynclient.Client
+}
+
+var _ FrameworkClient = &frameworkClient{}
+
+type FrameworkClient interface {
+	Get(gCtx goctx.Context, key dynclient.ObjectKey, obj runtime.Object) error
+	List(gCtx goctx.Context, opts *dynclient.ListOptions, list runtime.Object) error
+	Create(gCtx goctx.Context, obj runtime.Object, tCtx *TestCtx, cleanupOptions *CleanupOptions) error
```

Comment:
How about lumping in the testContext into the CleanupOptions since both of them are needed together to add a cleanup function to the test context.
```Go
type CleanupOptions struct {
	TestContext   *TestCtx
        Timeout       time.Duration
	RetryInterval time.Duration
        SkipPolling   bool
}
```

That way we can make Create less verbose:
```Go
Create(gCtx goctx.Context, obj runtime.Object, cleanupOptions *CleanupOptions)
```